pkg/dao/interfaces.go

```diff
@@ -44,24 +44,14 @@ func GetDaoRegistry(db *gorm.DB) *DaoRegistry {
 		},
 		TaskInfo:     taskInfoDaoImpl{db: db},
 		AdminTask:    adminTaskInfoDaoImpl{db: db, pulpClient: pulp_client.GetGlobalPulpClient()},
-		Domain:       domainDaoImpl{db: db},
+		Domain:       domainDaoImpl{db: db, pulpClient: pulp_client.GetGlobalPulpClient(), cpClient: candlepin_client.NewCandlepinClient()},
 		PackageGroup: packageGroupDaoImpl{db: db},
 		Environment:  environmentDaoImpl{db: db},
 		Template:     templateDaoImpl{db: db},
 	}
 	return &reg
 }
 
-// SetupGormTableOrFail this is necessary to enable soft-delete
-// on the deleted_at column of the template_repository_configurations table.
-// More info here: https://gorm.io/docs/many_to_many.html#Customize-JoinTable
-func SetupGormTableOrFail(db *gorm.DB) {
-	err := db.SetupJoinTable(models.Template{}, "RepositoryConfigurations", models.TemplateRepositoryConfiguration{})
-	if err != nil {
-		log.Logger.Fatal().Err(err).Msg("error setting up join table for templates_repository_configurations")
-	}
-}
```

Comment:
Was this causing that bug? Will soft delete still work without it?

Reply:
When i made the change here: https://github.com/content-services/content-sources-backend/pull/865/commits/3544d27bb0f3289d2ff5180457bea6fe41645e4c#diff-6daee329a68afa6ef518701e49dba065bb5372d88bec90794533ff924bf576d5R27  to link templates to TemplateRepositoryConfiguration directly, it complained.  My guess is this might have been why it wasn't working before?  (it was trying to link through that table to RepositoryConfigurations directly).  

what sort of behavior were you seeing?  Just the template_repository_configuration entry being deleted entirely instead of soft deleted?

Reply:
I suspect the change to the model may have made it okay to be removed. I noticed there was a comment above this function that said "we need this for soft delete to work", so I wanted to make sure you caught that.

You can see in the gorm docs, the models are arranged like what we previously had: https://gorm.io/docs/many_to_many.html#Customize-JoinTable

Reply:
i did a manual test by creating a template with 2 repos, I think ran the api server without any workers.  I then updated the template to only have 1 repo.  Finally I checked the db:
```
content=# select template_uuid, deleted_at from templates_repository_configurations ;
            template_uuid             |          deleted_at           
--------------------------------------+-------------------------------
 a00baafb-1d3e-49b0-b3e9-4aa21df87716 | 2024-11-08 19:16:08.869313+00
 a00baafb-1d3e-49b0-b3e9-4aa21df87716 | 

```

Finally i started the server again with the workers, and re-ran the query:

```
content=# select template_uuid, deleted_at from templates_repository_configurations 
;
            template_uuid             | deleted_at 
--------------------------------------+------------
 a00baafb-1d3e-49b0-b3e9-4aa21df87716 | 
```

so i think its good!